### PR TITLE
Add qcodes recipe

### PR DESCRIPTION
**Pull request by basnijholt**
Depends on `conda build==2.*`

### CODE REVIEW

File: recipes/qcodes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    nanoelectronic device experiments, it is not inherently limited 
+    to such experiments, and can be used anywhere a system with many 
+    degrees of freedom is controllable by computer.
+  doc_url: https://github.com/QCoDeS/Qcodes/blob/master/docs/
+  dev_url: https://github.com/QCoDeS/Qcodes
+
+extra:
+  recipe-maintainers:
+    - basnijholt
+    - AdriaanRol
```
Comment (synapticarbors) [inline recipes/qcodes/meta.yaml]: @AdriaanRol are you ok being listed as a co-maintainer?

File: recipes/qcodes/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - jupyter >=1.0.0
+    - ipywidgets >=4.1
+    - h5py >=2.6
+    - pathlib
+    - matplotlib
+    - coverage
+    - pyqtgraph
+
+test:
+  source_files:
```
Comment (synapticarbors) [inline recipes/qcodes/meta.yaml]: I believe `source_files` was introduced in conda-build 2.0.0, but currently conda-forge is using 1.21.14, which is causing an error. I'm not sure if there was some other mechanism for doing this in previous versions of conda-build. For now, you may just want to skip this step.

File: recipes/qcodes/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (synapticarbors) [inline recipes/qcodes/meta.yaml]: Remove comments here and elsewhere in the recipe.

File: recipes/qcodes/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  license_file: LICENSE.md
+  summary: 'Modular data acquisition and analysis framework'
+  description: |
+    QCoDeS is a Python-based data acquisition framework developed 
+    by the Copenhagen / Delft / Sydney / Microsoft quantum computing 
+    consortium. While it has been developed to serve the needs of 
+    nanoelectronic device experiments, it is not inherently limited 
+    to such experiments, and can be used anywhere a system with many 
+    degrees of freedom is controllable by computer.
+  doc_url: https://github.com/QCoDeS/Qcodes/blob/master/docs/
```
Comment (synapticarbors) [inline recipes/qcodes/meta.yaml]: I think the `doc_url` should be http://qcodes.github.io/Qcodes/

File: recipes/qcodes/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.zip
+  url: https://github.com/QCoDeS/Qcodes/archive/v{{ version }}.zip
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py2k]
```
Comment (synapticarbors) [inline recipes/qcodes/meta.yaml]: The `README` says this package is compatible with python 3.5+, but the `setup.py` pypi metadata indicates 3.3, 3.4, and 3.5 are supported. The docs say only 3.4 and 3.5.

File: recipes/qcodes/meta.yaml (hunk 1 above)
Comment (AdriaanRol) [inline recipes/qcodes/meta.yaml]: 👍

File: recipes/qcodes/meta.yaml (hunk 2 above)
Comment (basnijholt) [inline recipes/qcodes/meta.yaml]: I talked to one of the authors, he prefers to wait until it's fixed.

File: recipes/qcodes/meta.yaml (hunk 5 above)
Comment (basnijholt) [inline recipes/qcodes/meta.yaml]: Thanks for noting, @AdriaanRol do you know what it should be?

File: recipes/qcodes/meta.yaml (hunk 4 above)
Comment (basnijholt) [inline recipes/qcodes/meta.yaml]: Fixed in 89eddab.

File: recipes/qcodes/meta.yaml (hunk 5 above)
Comment (AdriaanRol) [inline recipes/qcodes/meta.yaml]: Made an issue in the QCoDeS repo to resolve this.

File: recipes/qcodes/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+#  source_files:
+#    - qcodes/test.py
+#  commands:
+#    - python qcodes/test.py
+  imports:
+    - qcodes
+
+about:
+  home: http://qcodes.github.io/Qcodes/
+  license: MIT
```
Comment (jakirkham) [inline recipes/qcodes/meta.yaml]: Could add `license_family: MIT`. Though it is optional.

File: recipes/qcodes/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.zip
+  url: https://github.com/QCoDeS/Qcodes/archive/v{{ version }}.zip
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [not py35]
```
Comment (jakirkham) [inline recipes/qcodes/meta.yaml]: So as to support Python 3.6 when it comes out. Could change `not py35` -> `py<35`.

**Comment by jakirkham** [comment]
Do we know what the intended Python compatibility is yet? Seems like that was unclear before. Otherwise this seems like a nice recipe ready to be merged.

**Comment by jakirkham** [comment]
Friendly nudge.

**Comment by synapticarbors** [comment]
@jakirkham was the nudge for me to do a final review and merge, or was it about something else?

**Comment by jakirkham** [comment]
Sorry, should be more specific. Am trying to understand the Python compatibility of this package. Do you or @basnijholt know?

**Comment by basnijholt** [comment]
I wouldn't know. @AdriaanRol could you comment?

**Comment by jakirkham** [comment]
Seems like the answer to @AdriaanRol's issue ( https://github.com/QCoDeS/Qcodes/issues/368 ) is [3.5 and above]( https://github.com/QCoDeS/Qcodes/commit/b9f41545aea7e01cf2d3283d66be18c675e89534 ). So we should be fine as is.

**Comment by jakirkham** [comment]
Thanks @basnijholt.

File: recipes/qcodes/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    - pathlib
+    - matplotlib
+    - coverage
+    - pyqtgraph
+
+test:
+#  source_files:
+#    - qcodes/test.py
+#  commands:
+#    - python qcodes/test.py
```
Comment (jakirkham) [inline recipes/qcodes/meta.yaml]: Should be possible to uncomment this in the feedstock as we now use `conda-build` 2.

**Pull request merged**